compiler/runtime/OMRRSSReport.hpp

```diff
@@ -0,0 +1,262 @@
+ * RSSItem can be added to RSSRegion. It is possible to associate
+ * debug counters with each item.
+ */
+class RSSItem
+   {
+   public:
+   TR_ALLOC(TR_Memory::Inliner);  // TODO: add new type
+   enum Type
+      {
+      unknown,
+      header,
+      alignment,
+      coldBlocks,
+      overEstimate,
+      OOL,
+      snippet
+      };
+
+   static const char *itemNames[];
+
+   RSSItem() : _type(unknown), _addr(NULL), _size(0), _counters(NULL) {}
+   RSSItem(Type type, uint8_t *addr, size_t size, TR_PersistentList<TR::DebugCounterAggregation> *counters)
+           : _type(type), _addr(addr), _size(size), _counters(counters){}
+   private:
+   enum Type _type;
+   uint8_t *_addr;
+   size_t _size;
+   TR_PersistentList<TR::DebugCounterAggregation> *_counters;
+   friend class RSSRegion;
+   friend class RSSReport;
+   };
+
+/*
+ * RSSRegion represents a region of memory that has starting address, size,
+ * name, and RSSItems associated with it. Regions can be added to RSSReport.
+ * Region's size can be updatee after it was added to the report.
+ * Regions can grow low to high, as well as high to low addresses.
+ */
+class RSSRegion
+   {
+   TR_ALLOC(TR_Memory::UnknownType);  // TODO: add new type
+public:
+   enum Grows
+      {
+      lowToHigh,
+      highToLow
+      };
+
+   RSSRegion(const char *name = NULL, uint8_t *start = 0, uint32_t size = 0,
+             Grows grows = lowToHigh, size_t pageSize = 0) :
+             _name(name), _start(start), _size(size), _grows(grows), _pageSize(pageSize),
+             _pageMap(TR::Compiler->persistentMemory()) { }
+
+   /**
+    * \brief
+    *  Sets page size
+    */
+   void setPageSize(size_t pageSize) { _pageSize = pageSize; }
+
+   /**
+    * \brief
+    *  Returns region's start
+    *
+    * \returns
+    *  Lowest address for regions that grow low to high and highest addres otherwise
+    */
+   uint8_t *getStart() {return _start;}
+
+   /**
+    * \brief Adds RSSItem to the region. If the item overlaps with any existing items
+    *        those items get removed.
+    *
+    * \param item
+    *        Item to add
+    * \param thredId
+    *        Thread id that adds the item
+    * \param methodName
+    *        Name of the method that adds the item
+    */
+   void addRSSItem(RSSItem *item, int32_t threadId, const char* methodName = NULL);
+
+   /**
+    * \brief
+    *      Adds item to a sorted list
+    *
+    * \param itemList
+    *        List to add to
+    * \param item
+    *        Item to add
+    */
+   void addToListSorted(TR_PersistentList<RSSItem> *itemList, RSSItem *item);
+
+   /**
+    * \brief
+    *        Returns list of RSSItems for the page given address belongs to
+    *
+    * \param address
+    *        Address
+    *
+    * \return
+    *        Items list
+    */
+   TR_PersistentList<RSSItem> getRSSItemList(uint8_t *address)
+         {
+         TR_ASSERT_FATAL(_pageSize >=0, "Page size should be set");
```

Comment:
Should be `_pageSize > 0`, otherwise it won't catch the case where the page size wasn't set (default is 0).

Reply:
Thanks for noticing. Of course, that was the whole idea!

Reply:
Fixed.